src/main/java/at/hannibal2/skyhanni/features/misc/MagicalPowerDisplay.kt

```diff
@@ -0,0 +1,106 @@
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.ProfileStorageData
+import at.hannibal2.skyhanni.events.InventoryOpenEvent
+import at.hannibal2.skyhanni.events.RenderItemTipEvent
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.test.command.ErrorManager
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.ItemCategory
+import at.hannibal2.skyhanni.utils.ItemUtils.getInternalNameOrNull
+import at.hannibal2.skyhanni.utils.ItemUtils.getItemCategoryOrNull
+import at.hannibal2.skyhanni.utils.ItemUtils.getItemRarityOrNull
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzRarity
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.NEUInternalName.Companion.asInternalName
+import at.hannibal2.skyhanni.utils.RegexUtils.matches
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import io.github.moulberry.notenoughupdates.util.stripControlCodes
+import net.minecraft.item.ItemStack
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+@SkyHanniModule
+object MagicalPowerDisplay {
+    private val config get() = SkyHanniMod.feature.inventory.magicalPower
+    private var contactAmount: Int
+        get() = ProfileStorageData.profileSpecific?.abiphoneContactAmount ?: -1
+        private set(value) { ProfileStorageData.profileSpecific?.abiphoneContactAmount = value }
+
+    /*
+    * REGEX-TEST: Accessory Bag
+    * REGEX-TEST: Accessory Bag (1/75)
+    * REGEX-TEST: Accessory Bag (909/394294)
+    * REGEX-TEST: Auctions Browser
+    * REGEX-TEST: Auctions: "ligma"
+    * REGEX-TEST: Auctions: ""sugoma""
+    * */
+    private val acceptedInvPattern by RepoPattern.pattern(
+        "inv.acceptable",
+        """^(Accessory Bag(?: \(\d+/\d+\))?|Auctions Browser|Manage Auctions|Auctions: ".*"?)$"""
+    )
+
+    @SubscribeEvent
+    fun onRenderItemTip(event: RenderItemTipEvent) {
+        if (!isEnabled()) return
+        if (!acceptedInvPattern.matches(InventoryUtils.openInventoryName().stripControlCodes())) return
+
+        val item = event.stack
+        val rarity = item.isAccessory() ?: return
+        val itemID = item.getInternalNameOrNull() ?: return
+
+        var endMP = rarity.toMP() ?: run {
+            ErrorManager.skyHanniError(
+                "Unknown rarity '$rarity' for item '${item.displayName}§7'"
+            )
+        }
+        if (itemID == "HEGEMONY_ARTIFACT".asInternalName())
+            endMP *= 2
+        if (itemID == "RIFT_PRISM".asInternalName())
+            endMP = 11
+        if (item.isAbicase())
+            endMP += contactAmount / 2
+        event.stackTip = "${if (config.colored) rarity.chatColorCode else "§7"}${endMP}"
+    }
+
+    @SubscribeEvent
+    fun onInventoryOpened(event: InventoryOpenEvent) {
+        if (!isEnabled()) return
+        if (event.inventoryName.startsWith("Abiphone")) {
```

Comment:
Use a repoPattern instead.

Reply:
Since Contact Directory is only available at XI Ultra Style and above I would need at least groups for

- 6 Ultra Style
- 7 Mega (Color)
- 8 Pro (Giga)
- Enormous
- Flip

Reply:
you can all that put into one single regex (`Abiphone .*`)

Reply:
But then the lower ones does not have and yet still get accepted

Reply:
Maybe we have missed something. The existing code is filled with string operations (startsWith, endsWith, contains, ==) on data we directly get from Hypixel. For maintaining compatibility with changed formatting on Hypixel's end, we want to change all those cases with regex via repo pattern. Since it's a ton of work to change the existing code at once, we start by trying to get rid of those patterns in new PRs as much as possible.

If you find an open pr where this is still the case, feel free to comment on it under.

Reply:
Wait are you talking about different abiphone names and not other prs?
in that case, please tell me what names are wrongly accepted and we can find a solution together

Reply:
In either case, the regex is checking the same thing as your startswith, so the existing logic has the same problems aready and this has nothing to do with using repo patterns over string operations

Reply:
Ok nvm I can just let through Abiphone and with current approach the book is checked instead. For lower type of phone the book will be null.